src/stack-unix/tcpip_stack_socket.ml

```diff
@@ -59,7 +61,8 @@ module V4 = struct
                  Lwt.return_unit) >>= fun () ->
             loop ()
           in
-          loop ())
+          Lwt.catch loop (fun _ -> Lwt.return_unit) >>= fun () ->
```

Comment:
It would be better to ignore just the `Canceled` exception, not all of them I think.